util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -91,178 +114,179 @@ func NewWorker() *Worker {
 
 // Start starts the worker.
 func (w *Worker) Start() {
-	w.stopCh = make(chan struct{})
-	// TODO(#1631): make controller and worker communicate using k8s API (with CustomResources)
-	// instead of http.
-	w.listenToController()
+	w.populatePodName()
+	w.populateDynamicResource()
+	w.watchForWork()
 }
 
-func (w *Worker) listenToController() {
-	handlers := handlersMap{
-		"/startTCPServer":  w.StartTCPServer,
-		"/startTCPClient":  w.StartTCPClient,
-		"/startUDPServer":  w.StartUDPServer,
-		"/startUDPClient":  w.StartUDPClient,
-		"/startHTTPServer": w.StartHTTPServer,
-		"/startHTTPClient": w.StartHTTPClient,
-		"/metrics":         w.Metrics,
+func (w *Worker) populatePodName() {
+	var isPodNameAvailable bool
+	w.podName, isPodNameAvailable = os.LookupEnv("POD_NAME")
+	if !isPodNameAvailable {
+		w.updateStatus(map[string]interface{}{"error": "Pod name not set as environment variable"})
```

Comment:
can you create a helper function:

makeErrorStatus(err error)

?

Reply:
Modified.